### PR TITLE
WIP: parse TileConfiguration.txt

### PR DESCRIPTION
**Pull request by dzenanz**
After commit 72d9ea6 I am running into:
`itkMontageTestOMC.obj : error LNK2019: unresolved external symbol "class std::vector<class std::vector<struct itk::Tile<2>,class std::allocator<struct itk::Tile<2> > >,class std::allocator<class std::vector<struct itk::Tile<2>,class std::allocator<struct itk::Tile<2> > > > > __cdecl itk::ParseTileConfiguration2D(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >)" (?ParseTileConfiguration2D@itk@@YA?AV?$vector@V?$vector@U?$Tile@$01@itk@@V?$allocator@U?$Tile@$01@itk@@@std@@@std@@V?$allocator@V?$vector@U?$Tile@$01@itk@@V?$allocator@U?$Tile@$01@itk@@@std@@@std@@@2@@std@@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@3@@Z) referenced in function "int __cdecl itkMontageTestOMC(int,char * * const)" (?itkMontageTestOMC@@YAHHQEAPEAD@Z)`

How to specify that Montage's tests should link to Montage library? Am I setting up the source part of the Module correctly?

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
-  COMPONENTS ITKImageIO ITKTransformIO Montage
-  )
+if(NOT ITK_SOURCE_DIR)
+  find_package(ITK REQUIRED COMPONENTS ITKImageIO ITKTransformIO Montage)
+else()
+  # when being built as part of ITK, ITKImageIO and ITKTransformIO
+  # lists of modules are not yet ready, causing a configure error
+  find_package(ITK REQUIRED COMPONENTS
+    Montage
+    ITKTestKernel # includes many I/O modules
```
Comment (thewtex) [inline examples/CMakeLists.txt]: `ITKImageIO` is preferred

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+set(Montage_SRC
+  itkParseTileConfiguration.cxx
+  )
+
+add_library(Montage ${Montage_SRC})
+# target_link_libraries(Montage ${ITKCommon}) # we do not link-depend on anything
+itk_module_target(Montage)
```
Comment (thewtex) [inline src/CMakeLists.txt]: here we want to use 

```
itk_module_add_library(Montage ${Montage_SRC})
```

per

https://github.com/InsightSoftwareConsortium/ITKModuleTemplate/blob/master/%7B%7Bcookiecutter.project_name%7D%7D/src/CMakeLists.txt

File: src/itkParseTileConfiguration.cxx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  tile.FileName = temp;
+  std::getline( ss, temp, ';' );
+  timePointID = temp;
+  std::getline( ss, temp, '(' );
+
+  using PointType = itk::Point< double, Dimension >;
+  PointType p;
+  for (unsigned d = 0; d < Dimension; d++)
+    {
+    ss >> p[d]; //TODO: use DoubleConversion here!
```
Comment (thewtex) [inline src/itkParseTileConfiguration.cxx]: yah :-)

File: src/itkParseTileConfiguration.cxx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  return row;
+}
+
+} // annonymous namespace
+
+
+
+namespace itk
+{
+Montage_EXPORT std::vector< std::vector< Tile< 2 > > >
```
Comment (thewtex) [inline src/itkParseTileConfiguration.cxx]: No export specification in the .cxx file -- only the .h file

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+set(Montage_SRC
+  itkParseTileConfiguration.cxx
+  )
```
Comment (thewtex) [inline src/CMakeLists.txt]: In the top level CMakeLists.txt file, we also need to add

```
set(Montage_LIBRARIES Montage
```

per https://github.com/InsightSoftwareConsortium/ITKModuleTemplate/blob/master/%7B%7Bcookiecutter.project_name%7D%7D/CMakeLists.txt#L4

File: examples/CMakeLists.txt (hunk 1 above)
Comment (dzenanz) [reply examples/CMakeLists.txt]: `ITKImageIO` does not work properly for remote examples when building them as part of ITK. I talked to Francois, and he says that is still WIP 😄

Comment (dzenanz) [reply examples/CMakeLists.txt]: There is explanation for it just a few lines above.

**Comment by dzenanz** [comment]
I will make a new PR when this is ready for merging.

**Pull request closed without merging**